### PR TITLE
Add linters and refactor code for maintainability and code reuse

### PR DESCRIPTION
**Pull request by tumido**
Resolves: #47 #46 

- Adds new dependencies for dev environment to support hot reloading via `nodaemon` and `ts-node`
- Extends the Probot class with Kubernetes client and `updateToken` method (these methods now also use `app.log`) Moved to separate file from the handlers.
- Update handlers to use Probot app methods
- Adds Prettier and Eslint config for code linting and formatting
- Adds `npm run lint` to highlight violations
- Adds `npm run dev` for dev env with hot reloading
- Adds NVM config to match dev environment NodeJS version
- Simplified logging, since [Probot log uses Pino](https://probot.github.io/docs/logging/), which already contains most of the information passed to our log lines like installation ID...
- Fixes a case when the org has no repository present - instead of parsing `payload.repositories`, we use `payload.installation.account.login`

~Note: Pre-commit may fail if the CI runner doesn't have node installed in its runtime (Eslint hook requires it)~

### CODE REVIEW

**Comment by tumido** [comment]
Hey, I know this PR is huge, sorry for that. Most of it is moving the code around and letting Prettier to format it. I had to move the content from `index.ts` into `app.ts` so we can support hot-reloading via `index.ts`. Please review

/cc @harshad16 @mimotej @SamoKopecky @KPostOffice

File: README.md [hunk 1]
```diff
@@ -53,26 +53,34 @@ Reference: [docs](https://probot.github.io/docs/configuration/)
...
+      ```
 
-    ```sh
-        # Install dependencies
-        npm install
+      Note: `nvm use` can be called automatically by your shell on entering the directory, see docs on [NVM](https://github.com/nvm-sh/nvm#deeper-shell-integration) or use a plugin like [oh-my-zsh nvm](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/nvm)
 
-        # Build all the requirements
-        npm run build
+   2. Remember to set environment variables via `.env` file. We require:
```
Comment (SamoKopecky) [inline README.md:70]: Don't we also need to set the `PRIVATE_KEY` variable?

**Review by SamoKopecky — APPROVED** [review]
Just have one small question about the dev setup, otherwise LGTM!

File: README.md (hunk 1 above)
Comment (tumido) [reply README.md:70]: Not necessarily see:
https://probot.github.io/docs/development/#manually-configuring-a-github-app

image

However we should specify that the priv key has to be located in the folder... I'll update the PR.

**Review by harshad16 — APPROVED** [review]
lgtm
i like the changes, seems very neat.
:+1:

**Pull request merged**